libs/langchain/langchain/chains/combine_documents/map_reduce.py

```diff
@@ -1,18 +1,197 @@
 """Combining documents by mapping a chain over them first, then combining results."""
-
 from __future__ import annotations
 
+from functools import partial
 from typing import Any, Dict, List, Optional, Tuple, Type
 
 from langchain_core.documents import Document
+from langchain_core.language_models import LanguageModelLike
+from langchain_core.output_parsers import StrOutputParser
+from langchain_core.prompts import BasePromptTemplate, PromptTemplate
 from langchain_core.pydantic_v1 import BaseModel, Extra, create_model, root_validator
+from langchain_core.runnables import Runnable, RunnablePassthrough
 from langchain_core.runnables.config import RunnableConfig
 
 from langchain.callbacks.manager import Callbacks
-from langchain.chains.combine_documents.base import BaseCombineDocumentsChain
+from langchain.chains.combine_documents.base import (
+    DOCUMENTS_KEY,
+    BaseCombineDocumentsChain,
+    format_document_inputs,
+    format_document_inputs_as_list,
+    validate_prompt,
+)
 from langchain.chains.combine_documents.reduce import ReduceDocumentsChain
 from langchain.chains.llm import LLMChain
 
+""" --- LCEL Runnable chains --- """
+
+
+def create_map_documents_chain(
+    llm: LanguageModelLike,
+    prompt: BasePromptTemplate,
+    *,
+    document_prompt: Optional[BasePromptTemplate] = None,
+) -> Runnable[Dict[str, Any], List[Document]]:
+    """Create a chain that updates the contents of a list of documents by passing them to a model.
```

Comment:
`Create a chain that passes each document to an LLM and then creates a new list of Documents` - tighter?